### PR TITLE
implement configurable form OData feed

### PR DESCRIPTION
**Pull request by nickpell**
similar to https://github.com/dimagi/commcare-hq/pull/24686

### CODE REVIEW

File: corehq/apps/api/odata/views.py [hunk 1]
```diff
@@ -141,6 +142,46 @@ def get(self, request, domain):
...
+
+    urlname = 'odata_form_metadata_from_export_instance'
+
+    @method_decorator(basic_auth_or_try_api_key_auth)
+    @method_decorator(require_permission(Permissions.edit_data, login_decorator=None))
+    @method_decorator(toggles.ODATA.required_decorator())
+    def get(self, request, domain):
+        configs = get_odata_form_configs_by_domain(domain)
+        config_ids_to_properties = OrderedDict()
+        for config in sorted(configs, key=lambda _config: _config.get_id):  # For deterministic tests
```
Comment (orangejenny) [inline corehq/apps/api/odata/views.py]: Is it possible for the sorting to happen in the tests instead of here?

**Review by orangejenny — APPROVED** [review]

File: corehq/apps/api/odata/views.py (hunk 1 above)
Comment (nickpell) [reply corehq/apps/api/odata/views.py]: good point: https://github.com/dimagi/commcare-hq/pull/24770

File: corehq/apps/api/odata/serializers.py [hunk 2]
```diff
@@ -175,3 +176,55 @@ def serialize_cases_using_config(cases, config):
             }
             for case_data in cases
         ]
+
+
+class ODataFormSerializer(Serializer):
```
Comment (esoergel) [inline corehq/apps/api/odata/serializers.py]: This is nearly identical to `ODataCaseSerializer` except for the id field and the view it redirects to.  Do you have plans to deduplicat those?  Do you expect further variation to arise?

File: corehq/apps/api/odata/tests/test_serializers.py [hunk 3]
```diff
@@ -239,3 +241,97 @@ def test_next_link_absent(self):
...
+                                            PathNode(name='user_id')
+                                        ]
+                                    ),
+                                    selected=True,
+                                )
+                            ]
+                        )
+                    ]
+                )
+            ),
```
Comment (esoergel) [inline corehq/apps/api/odata/tests/test_serializers.py]: These tests would be a little easier to read if you pulled the instance initialization out into a helper

File: corehq/apps/api/resources/v0_5.py [hunk 4]
```diff
@@ -1043,3 +1044,52 @@ def prepend_urls(self):
...
+        }
+
+    class Meta(v0_4.XFormInstanceResource.Meta):
+        authentication = ODataAuthentication(Permissions.edit_data)
+        resource_name = 'odata/forms'
+        serializer = ODataFormSerializer()
+        limit = 2000
+        max_limit = 10000
+
+    def prepend_urls(self):
```
Comment (esoergel) [inline corehq/apps/api/resources/v0_5.py]: is this some tastypie thing to let you create a custom API URL?

Comment (nickpell) [reply corehq/apps/api/resources/v0_5.py]: exactly

File: corehq/apps/api/odata/serializers.py (hunk 2 above)
Comment (nickpell) [reply corehq/apps/api/odata/serializers.py]: I'm not sure if further variation will arise.  I'd probably recommend considering deduplication after the beta has started.

**Pull request merged**